### PR TITLE
Increase the scope of actions for the strategist

### PR DESCRIPTION
**Pull request by franckc**
Implement #563 

Note: The strategist address is currently not stored on the strategies. Which means we currently can't just add a `onlyStrategist` modifier on the strategies methods. I considered  two options:
  1. Add a slot to the strategies contract to store the strategist address. Involves redeploying all the strategies and set the strategist address on each of them. Also if in the future, if we change the strategist address, we would have to make sure to set it not only on the vault but also on all the strategies.
  2. Have the vault call the strategy's method, on behalf of the strategist. Involves redeploying VaultAdmin. Adds a bit of cruft to VaultAdmin.
In the end, I'm favoring option 2. since it seemed to me the simplest but let me know any feedback...

If you made a contract change, make sure to complete the checklist below before merging it in master.

Contract change checklist:
  - [ ] Code reviewed by 2 reviewers. See [template](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review.md) and [example](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review-Example.md).
  - [X] Unit tests pass
  - [X] Slither tests pass with no warning
  - [ ] Echidna tests pass if PR includes changes to OUSD contract (not automated, run manually on local)

### CODE REVIEW

File: contracts/test/vault/index.js [hunk 1]
```diff
@@ -531,4 +539,35 @@ describe("Vault", function () {
...
+      vault,
+      governor,
+      strategist,
+      compoundStrategy,
+      matt,
+    } = await loadFixture(defaultFixture);
+    await vault.connect(governor).approveStrategy(compoundStrategy.address);
+    await vault
+      .connect(governor)
+      .withdrawAllFromStrategy(compoundStrategy.address);
```
Comment (tomlinton) [inline contracts/test/vault/index.js:565]: Maybe add some assets to the Compound strategy and ensure they get withdrawn?

**Review by tomlinton — APPROVED** [review]
Looks good, one optional comment about expanding the tests

File: contracts/test/vault/index.js (hunk 1 above)
Comment (franckc) [reply contracts/test/vault/index.js:565]: Still on my TODO list. I did not forget...

**Comment by franckc** [comment]
@DanielVF Any feedback on the PR or can I merge it to master and make it part of the next mainnet deploy?

**Review by DanielVF — APPROVED** [review]
👍

**Pull request merged**